insights/specs/datasources/yum_updates.py

```diff
@@ -18,83 +16,176 @@ def sorted_cmp(it, cmp):
+                    updates_list.append(upd)
+                return nevra, updates_list
+
+            @staticmethod
+            def pkg_nevra(pkg):
+                return "{}-{}:{}-{}.{}".format(pkg.name, pkg.epoch, pkg.version,
+                                               pkg.release, pkg.arch)
+
+            @staticmethod
+            def pkg_repo(pkg):
+                return pkg.reponame
+
+            @staticmethod
+            def advisory(pkg):
+                errata = pkg.get_advisories(hawkey.EQ)
+                return errata[0].id if len(errata) > 0 else None
+
+            def last_update(self):
+                last_ts = 0
+                for repo in self.base.repos.iter_enabled():
+                    repo_ts = repo._repo.getTimestamp()
+                    if repo_ts > last_ts:
+                        last_ts = repo_ts
+                return last_ts
+
+    except ImportError:
+
         import yum
 
-        self.base = yum.YumBase()
-        self.base.doGenericSetup(cache=1)
-        self.releasever = self.base.conf.yumvar['releasever']
-        self.basearch = self.base.conf.yumvar['basearch']
-        self.packages = []
-        self.repos = []
-        self.updict = {}
-
-    def __enter__(self):
-        return self
-
-    def __exit__(self, *args):
-        pass
-
-    @staticmethod
-    def pkg_cmp(a, b):
-        vercmp = a.verCMP(b)
-        if vercmp != 0:
-            return vercmp
-        if a.repoid != b.repoid:
-            return -1 if a.repoid < b.repoid else 1
-        return 0
-
-    def sorted_pkgs(self, pkgs):
-        return sorted_cmp(pkgs, self.pkg_cmp)
-
-    def load(self):
-        self.base.doRepoSetup()
-        self.base.doSackSetup()
-        self.packages = self.base.pkgSack.returnPackages()
-        self.repos = self.base.repos.repos
-        self._build_updict()
-
-    def _build_updict(self):
-        self.updict = {}
-        for pkg in self.packages:
-            self.updict.setdefault(pkg.na, []).append(pkg)
-
-    def enabled_repos(self):
-        return [repo.id for repo in self.base.repos.listEnabled()]
-
-    def installed_packages(self):
-        return self.base.rpmdb.returnPackages()
-
-    def updates(self, pkg):
-        nevra = pkg.nevra
-        updates_list = []
-        for upg in self.updict.get(pkg.na, []):
-            if upg.verGT(pkg):
-                updates_list.append(upg)
-        return nevra, updates_list
-
-    @staticmethod
-    def pkg_nevra(pkg):
-        return "{}-{}:{}-{}.{}".format(pkg.name, pkg.epoch, pkg.version, pkg.release, pkg.arch)
-
-    @staticmethod
-    def pkg_repo(pkg):
-        return pkg.repoid
-
-    def advisory(self, pkg):
-        adv = self.base.upinfo.get_notice(pkg.nvr)
-        if adv:
-            return adv.get_metadata()['update_id']
-        return None
-
-    @staticmethod
-    def last_update():
-        return 0
-
-
-@datasource(HostContext, [IsRhel7])
```

Comment:
Just curious why you removed the datasource decorator instead of just removing the IsRhel7 part?

Reply:
The datasource decorator was added in the past to constrain this datasource to RHEL7 only so I removed it because now we can run it on all systems.
But it turned out it's wrong and need to add `IsRhel8, IsRhel9` instead of removing it ;-).

Reply:
Ah yeah the decorator is required for every datasource.